core/src/main/java/io/aiven/klaw/service/TopicOverviewService.java

```diff
@@ -0,0 +1,529 @@
+  }
+
+  private void enrichTopicOverview(
+      int tenantId,
+      List<Topic> topics,
+      TopicOverview topicOverview,
+      String syncCluster,
+      List<TopicInfo> topicInfoList,
+      List<TopicHistory> topicHistoryList) {
+    ArrayList<TopicHistory> topicHistoryFromTopic;
+    for (Topic topic : topics) {
+      TopicInfo topicInfo = new TopicInfo();
+      topicInfo.setCluster(getEnvDetails(topic.getEnvironment(), tenantId).getName());
+      topicInfo.setClusterId(topic.getEnvironment());
+      topicInfo.setNoOfPartitions(topic.getNoOfPartitions());
+      topicInfo.setNoOfReplcias(topic.getNoOfReplcias());
+      topicInfo.setTeamname(manageDatabase.getTeamNameFromTeamId(tenantId, topic.getTeamId()));
+
+      if (syncCluster != null && syncCluster.equals(topic.getEnvironment())) {
+        topicOverview.setTopicDocumentation(topic.getDocumentation());
+        topicOverview.setTopicIdForDocumentation(topic.getTopicid());
+      }
+
+      if (topic.getHistory() != null) {
+        try {
+          topicHistoryFromTopic =
+              OBJECT_MAPPER.readValue(topic.getHistory(), new TypeReference<>() {});
+          topicHistoryList.addAll(topicHistoryFromTopic);
+        } catch (JsonProcessingException e) {
+          log.error("Unable to parse topicHistory ", e);
+        }
+      }
+      topicInfoList.add(topicInfo);
+    }
+
+    if (topicOverview.getTopicIdForDocumentation() == null) {
+      topicOverview.setTopicDocumentation(topics.get(0).getDocumentation());
+      topicOverview.setTopicIdForDocumentation(topics.get(0).getTopicid());
+    }
+
+    topicOverview.setTopicHistoryList(topicHistoryList);
+  }
+
+  private List<AclInfo> applyFiltersAclsForSOT(
+      Integer loggedInUserTeam, List<Acl> aclsFromSOT, int tenantId) {
+
+    List<AclInfo> aclList = new ArrayList<>();
+    AclInfo mp;
+
+    for (Acl aclSotItem : aclsFromSOT) {
+      mp = new AclInfo();
+      mp.setEnvironment(aclSotItem.getEnvironment());
+      mp.setEnvironmentName(getEnvDetails(aclSotItem.getEnvironment(), tenantId).getName());
+      mp.setTopicname(aclSotItem.getTopicname());
+      mp.setAcl_ip(aclSotItem.getAclip());
+      mp.setAcl_ssl(aclSotItem.getAclssl());
+      mp.setTransactionalId(aclSotItem.getTransactionalId());
+      mp.setTeamname(manageDatabase.getTeamNameFromTeamId(tenantId, aclSotItem.getTeamId()));
+      mp.setConsumergroup(aclSotItem.getConsumergroup());
+      mp.setTopictype(aclSotItem.getTopictype());
+      mp.setAclPatternType(aclSotItem.getAclPatternType());
+      mp.setReq_no(aclSotItem.getReq_no() + "");
+      if (aclSotItem.getTeamId() != null && aclSotItem.getTeamId().equals(loggedInUserTeam))
+        mp.setShowDeleteAcl(true);
+
+      if (aclSotItem.getAclip() != null || aclSotItem.getAclssl() != null) aclList.add(mp);
+    }
+    return aclList;
+  }
+
+  private List<Acl> getAclsFromSOT(
+      String env, String topicNameSearch, boolean regex, int tenantId) {
+    List<Acl> aclsFromSOT;
```

Comment:
probably it's better to use `set` since there are `contains` checks

Reply:
The underlying repo method returns a list, and changing that has a bigger impact, and have other method calls on that list. It has a bigger impact I see. May be it's ok here to keep it as is.

Reply:
ok, may be could be done separately
btw here it is probably a related PR https://github.com/aiven/klaw/pull/231 to switch to sets

Reply:
thats great, thanks.